src/Encoder.c

```diff
@@ -0,0 +1,4344 @@
+ *
+ * @return  A zyan status code.
+ */
+ZyanBool ZydisArePrefixesCompatible(const ZydisEncoderInstructionMatch *match)
+{
+    if ((!match->base_definition->accepts_segment) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_SEGMENT_MASK))
+    {
+        return ZYAN_FALSE;
+    }
+    if (match->definition->encoding != ZYDIS_INSTRUCTION_ENCODING_LEGACY)
+    {
+        return (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_PREFIX_MASK) ? ZYAN_FALSE : ZYAN_TRUE;
+    }
+
+    const ZydisInstructionDefinitionLEGACY *legacy_def = 
+        (const ZydisInstructionDefinitionLEGACY *)match->base_definition;
+    if ((!legacy_def->accepts_LOCK) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_LOCK))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_REP) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_REP))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_REPEREPZ) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_REPEREPZ))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_REPNEREPNZ) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_REPNEREPNZ))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_BOUND) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_BND))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_XACQUIRE) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_XACQUIRE))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_XRELEASE) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_XRELEASE))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_branch_hints) &&
+        (match->attributes & (ZYDIS_ENCODABLE_ATTRIBUTE_BRANCH_NOT_TAKEN | 
+                              ZYDIS_ENCODABLE_ATTRIBUTE_BRANCH_TAKEN)))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_NOTRACK) &&
+        (match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_NOTRACK))
+    {
+        return ZYAN_FALSE;
+    }
+    if ((!legacy_def->accepts_hle_without_lock) &&
+        (match->attributes & (ZYDIS_ENCODABLE_ATTRIBUTE_XACQUIRE | 
+                              ZYDIS_ENCODABLE_ATTRIBUTE_XRELEASE)) &&
+        !(match->attributes & ZYDIS_ENCODABLE_ATTRIBUTE_LOCK))
+    {
+        return ZYAN_FALSE;
+    }
+
+    return ZYAN_TRUE;
+}
+
+/**
+ * Returns operand mask containing information about operand count and types in a compressed form.
+ *
+ * @param   request     A pointer to `ZydisEncoderRequest` struct.
+ *
+ * @return  Operand mask.
+ */
+ZyanU16 ZydisGetOperandMask(const ZydisEncoderRequest *request)
+{
+    ZyanU16 operand_mask = request->operand_count;
+    ZyanU8 bit_offset = 3;
+    for (ZyanU8 i = 0; i < request->operand_count; ++i)
+    {
+        operand_mask |= (request->operands[i].type - ZYDIS_OPERAND_TYPE_REGISTER) << bit_offset;
+        bit_offset += 2;
```

Comment:
I will make this change but also add an assert with raw numerical value to act as a reminder that consistency with generator is required here.

Reply:
Sounds good! `ZYAN_STATIC_ASSERT` should work here, making this a compiler time detection (in case you didn't stumble across this lesser used macro yet).